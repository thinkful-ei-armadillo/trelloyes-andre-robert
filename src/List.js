import React, { Component } from 'react';
import './List.css';
import Card from './Card';

class List extends Component {
  render() {
    return (
      <section className="List">
        <header className="List-header">
          <h2>{this.props.header}</h2>
        </header>
        <div className="List-cards">
          {this.props.cards.map((card, index) => <Card key={index} title={card.title} content={card.content} />)}
        </div>
      </section>
    )
  }
}

export default List;
