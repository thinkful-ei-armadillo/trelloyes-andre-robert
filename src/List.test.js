import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';
import {STORE} from './store'

describe('Card component testing', () => {
  // smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List header="Test" cards={[STORE.allCards.a, STORE.allCards.c, STORE.allCards.f]} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<List header="Test" cards={[STORE.allCards.a,STORE.allCards.c,STORE.allCards.f]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
 });