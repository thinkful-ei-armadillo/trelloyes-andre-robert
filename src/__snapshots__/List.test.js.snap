// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component testing renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Test
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Third card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Sixth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
